collection/library/configuredinterface_snitch

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+
+import glob
+import hashlib
+from os import listdir
+from os import path
+from os.path import isfile, join
+import re
+import StringIO
+
+
+DOCUMENTATION = '''
+---
+module: configuredinterface_snitch
+
+short_description: Gathers configured values from network interface files.
+
+version_added: "2.1.6"
+
+description:
+    - "Pulls network interface files from a host"
+    - "Parses the files and returns a dict of key and values for configured keys."
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - Sudarshan Acharya
+'''
+
+RETURN = '''
+payload:
+    description: |
+       Dict of configured keys and their values.
+    type: dict
+doctype:
+    description: Type of document. Will always be 'configuredinterface'
+    type: str
+'''
+
+
+CONFIGURED_PARAMS = [
+    "mtu",
+    "offload-sg",
+    "bridge_waitport",
+    "bridge_fd",
+    "bridge_ports",
+    "bridge_stp",
+    "address",
+    "netmask",
+    "dns-nameservers",
+    "gateway"
+]
+
+
+NETWORK_INTERFACE_FILENAME="/etc/network/interfaces"
+
+
+class InterfaceParser:
+
+    def __init__(self):
+        self.directives = []
+
+        content=[]
+        self.source_files=self.getSources()
+        for source_file in self.source_files:
+            fd=open(source_file)
+            for line in fd.read().splitlines():
+                line=line.strip()
+                content.append(line)
+            fd.close()
+        self.iterator=iter(content)
+        self.parse()
+
+    def parse(self):
+        for directive in self.iterator:
+            while directive.startswith("iface"):
+                sup=None
+                subs=[]
+                for sub in self.iterator:
+                    if sub.startswith(("auto","iface","source")):
+                        sup = sub
+                        break
+                    else:
+                        subs.append(sub)
+                if subs:
+                    self.directives.append([directive, subs])
+                else:
+                    self.directives.append([directive])
+                if not sup:
+                    return
+                directive = sup
+            self.directives.append([directive])
+        return
+
+    def getSources(self):
```

Comment:
https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables
Method names should be lower case with underscores